pyccel/cuda/cuda_thread_indexing.py

```diff
@@ -0,0 +1,80 @@
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+"""
+This module contains all the CUDA thread indexing methods
+"""
+class CudaThreadIndexing:
+    """
+    Class representing the CUDA thread indexing.
+
+    Class representing the CUDA thread indexing.
+    """
+    def __init__(self, block_idx, thread_idx):
+        self._block_idx = block_idx
+        self._thread_idx = thread_idx
+
+    def threadIdx(self, dim):
+        """
+        Get the thread index.
+
+        Get the thread index.
+
+        Parameters
+        ----------
+        dim : int
+            The dimension of the indexing. It can be:
+            - 0 for the x-dimension
+            - 1 for the y-dimension
+            - 2 for the z-dimension
+
+        Returns
+        -------
+        int
+            The index of the thread in the specified dimension of its block.
+        """
+        return self._thread_idx
+
```

Comment:
/bot accept
this will be tested https://github.com/pyccel/pyccel-cuda/pull/66

Reply:
Can these be tested with a print? The analysis of the results will have to be done slightly differently but it should be possible.
E.g you could do one test for block_idx and one for thread_idx:
```python
from pyccel.decorators import kernel
from pyccel            import cuda

@kernel
def print_block():
    print(cuda.block_idx(0))

def f():
    print_block[5,5]()
    cuda.synchronize()

if __name__ == '__main__':
    f()
```
Instead of comparing strings, you would have to split the output into integers and check that all expected integers are present. E.g:
```python
python_block_idx = [int(word) for word in python_output]
for i in range(5):
    assert python_block_idx.count(i) == 5
```

Reply:
Even if the translation is not yet available it should be possible to test this with a Python only test

Reply:
Do we have a way to add tests like that to Pyccel ??

Reply:
I'm not sure if you really need that, Emily, and if there's a way to test it without going beyond the scope of this PR. I don't see a need to test thread indexing without arrays, as testing with arrays should be easier and more relevant. and the PR for arrays depends on this one

Reply:
>  if there's a way to test it without going beyond the scope of this PR

I am not sure how adding a test to ensure that the Python code gives the output that is expected after this PR is going beyond the scope of this PR?

> I don't see a need to test thread indexing without arrays

If thread indexing doesn't make sense without arrays then why do we need this PR?

> as testing with arrays should be easier

Are the 3 lines that I suggested complicated?

>  and the PR for arrays depends on this one

In that case we should be sure that this one works as intended. And we can't be sure without a test

Reply:
Does that look better, Emily  ded14a7 ??

Reply:
Yes. Great. Thanks :)